### PR TITLE
added customization closures

### LINKED ISSUE #25: [Bug]: Error in timeline view for User.
### What happened?

Im getting this error when i try to view the timeline table action.

`Rmsramos\Activitylog\Infolists\Components\TimeLineTitleEntry::getModifiedState(): Return value must be of type ?Illuminate\Support\HtmlString, string returned`

image

### How to reproduce the bug

Has existing data generated in previous logger.

### Package Version

1.0.2

### PHP Version

8.3.3

### Laravel Version

11.10.0

### Which operating systems does with happen with?

_No response_

### Notes

_No response_

### PR DESCRIPTION
**Pull request by ainesophaur**
- Added the ability to modify the eloquent query via closure  
- Added the ability to provide the list of activities via closure
- Added the ability to set the text/value for `TimeLineTitleEntry` via closure

all of the closures are backed by the `evaluate()` function so you're able to inject the regular filament variables like `$record`, `$component` etc

```php
ActivityLogTimelineAction::make('Activities')
    ->modifyQueryUsing(function (Builder $query) { 
    //some logic modify the query
    // ex return $query->where('log_name', '=', 'custom');
    })
```

```php
ActivityLogTimelineAction::make('Activities')
    ->activitiesUsing(function (Model $record) { 
    //some logic to fetch the activities as a collection
    // ex Activity::all() or $record->activities;
    })
```

```php
ActivityLogTimelineAction::make('Activities')
 ->shouldModifyTitleUsing(function (Activity $record) {
        //by default returns true, allows user to selectively skip the custom modifyTitleUsing function
        return $record->description !== $record->event;
    })
  ->modifyTitleUsing(function (Activity $record) {
      return $record->description;
  })
```

- fixes #25. [HasModifiedState getModifiedState](https://github.com/rmsramos/activitylog/blob/b8b3426c911d7d0a823a8d08261340d517251cba/src/Infolists/Concerns/HasModifyState.php#L19)  is typed to return `HtmlString` but  [TimeLineTitleEntry](https://github.com/rmsramos/activitylog/blob/b8b3426c911d7d0a823a8d08261340d517251cba/src/Infolists/Components/TimeLineTitleEntry.php#L53) can return a regular string if the description does not match the event.

### CODE REVIEW

**Comment by rmsramos** [comment]
It looks good, at the weekend I'll take my time to launch a new release.
thanks for the contribution

File: src/Actions/ActivityLogTimelineAction.php [hunk 1]
```diff
@@ -128,29 +161,70 @@ public function getLimit(): ?int
...
+    public function modifyQueryUsing(Closure $closure): static
+    {
+        $this->modifyQueryUsing = $closure;
+        return $this;
+    }
+
+    public function getModifyQueryUsing(Builder $builder): Builder
+    {
+        $this->evaluate($this->modifyQueryUsing, ['builder' => $builder]);
+        return  $builder;
```
Comment (sparksp) [inline src/Actions/ActivityLogTimelineAction.php:184]: This returns the original `$builder` not the modified query.

File: src/Actions/ActivityLogTimelineAction.php [hunk 2]
```diff
@@ -128,29 +161,70 @@ public function getLimit(): ?int
...
+
+    protected function getActivities(?Model $record, ?array $relations = null): Collection
+    {
+        if($activities = $this->getActivitiesUsing()) {
+            return $activities;
+        } else {
+            $builder = $this->getQuery()
+                ->latest()
+                ->limit($this->getLimit());
+            $this->getModifyQueryUsing($builder);
```
Comment (sparksp) [inline src/Actions/ActivityLogTimelineAction.php:221]: The modified query is not used.

**Review by sparksp — COMMENTED** [review]
I have noticed that `modifyQueryUsing` does not get correctly applied.

**Pull request merged**